encrypt-decrypt_test.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"testing"
+)
+
+func TestBase64_md5hash(t *testing.T) {
+	testCases := []struct {
+		input    []byte
+		expected string
+	}{
+		{[]byte("hello"), "XUFAKrxLKna5cZ2REBfFkg=="},
+		{[]byte(""), "1B2M2Y8AsgTpgAmY7PhCfg=="}, // MD5 hash of an empty string
+		{[]byte("a longer string with spaces"), "h/BochuK2O56DxYHJbWijA=="},
+	}
+
+	for _, tc := range testCases {
+		actual := Base64MD5Hash(tc.input)
+		if actual != tc.expected {
+			t.Errorf("For input %q, expected hash %q but got %q", tc.input, tc.expected, actual)
+		}
+	}
+}
+
+
+//Since it uses Google Cloud KMS, integration tests needed. But implemented test using static data and context.
+func TestEncryptBytes(t *testing.T) {
+	ctx := context.Background()
+	resourceName := "projects/cmetestproj/locations/global/keyRings/gcsproxytest/cryptoKeys/gcsproxy"
```

Comment:
For unit test, external resource shouldn't be dependent on . It's likely that the AEAD has to be mocked.